### PR TITLE
feat: rewrite e2e test(route_with_vars_test) with ginkgo

### PR DESCRIPTION
**Pull request by Jaycean**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#1500

___
### New feature or improvement
-  Rewrite e2e test(route_with_vars_test) with ginkgo

### CODE REVIEW

File: api/test/e2enew/route/route_with_vars_test.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+		})
+	})
+	ginkgo.It("hit the route with right args", func() {
+		base.RunTestCase(base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         `/hello`,
+			Query:        "name=aaa",
+			ExpectStatus: http.StatusOK,
+			Sleep:        base.SleepTime,
```
Comment (starsz) [inline api/test/e2enew/route/route_with_vars_test.go:68]: Maybe we can check the response body?

File: api/test/e2enew/route/route_with_vars_test.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+		})
+	})
+	ginkgo.It("hit the route with right header", func() {
+		base.RunTestCase(base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"k": "header"},
+			Path:         `/hello`,
+			ExpectStatus: http.StatusOK,
+			Sleep:        base.SleepTime,
```
Comment (starsz) [inline api/test/e2enew/route/route_with_vars_test.go:118]: Ditto.

File: api/test/e2enew/route/route_with_vars_test.go [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+			Path:         "/apisix/admin/routes/r1",
+			Body:         string(_createRouteBody),
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		})
+	})
+
+	ginkgo.It("verify route", func() {
+		base.RunTestCase(base.HttpTestCase{
+
```
Comment (starsz) [inline api/test/e2enew/route/route_with_vars_test.go:306]: redundant line.

**Review by nic-chen — APPROVED** [review]

File: api/test/e2enew/route/route_with_vars_test.go (hunk 1 above)
Comment (Jaycean) [reply api/test/e2enew/route/route_with_vars_test.go:69]: done.

File: api/test/e2enew/route/route_with_vars_test.go (hunk 2 above)
Comment (Jaycean) [reply api/test/e2enew/route/route_with_vars_test.go:122]: done.

File: api/test/e2enew/route/route_with_vars_test.go (hunk 3 above)
Comment (Jaycean) [reply api/test/e2enew/route/route_with_vars_test.go]: done.

File: api/test/e2enew/route/route_with_vars_test.go [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+		base.RunTestCase(base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         `/hello`,
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   `{"error_msg":"404 Route Not Found"}`,
+			Sleep:        base.SleepTime,
+		})
+	})
+
```
Comment (starsz) [inline api/test/e2enew/route/route_with_vars_test.go:93]: Redundant line.

Comment (Jaycean) [reply api/test/e2enew/route/route_with_vars_test.go]: done

**Comment by Jaycean** [comment]
cc @starsz  PTAL Thks.

**Comment by starsz** [comment]
> cc @starsz PTAL Thks.

LGTM.Thank you very much. : )

**Review by starsz — APPROVED** [review]

**Pull request merged**